README.md

```diff
@@ -5,45 +5,164 @@
+        key_out:Optional[str]=None,
+        ):
+
+        #we extract the element in the defined key location (for example 'input.xray_img')
+        img = sample_dict[key_in] 
+        assert isinstance(img, np.ndarray), f'Expected np.ndarray but got {type(img)}'
+        processed_img = np.pad(img, pad_width=padding, mode=mode, constant_values=fill)         
+        
+        #store the result in the requested output key (or in key_in if no key_out is provided)
+        key_out = key_in if key_out is None
+        sample_dict[key] = processed_img 
+
+        #returned the modified nested dict
+        return sample_dict
 ```
+
+Since the key location isn't hardcoded, this module can be eaily reused across different research projects with very different data sample structures. More code reuse - Hooray!  
+  
+FuseMedML-style components in general are any classes or functions that define which key paths will be written and which will be read.
+Arguments can be freely named, and you don't even have to write anything to the nested dict.
+Some FuseMedML components return a value directly - for example, loss functions.
+
+## 2. "Batteries included" key components, built using the same design concept
+
+### **fuse.data** - A **declarative** super flexible data processing pipeline
+* Easy dealing with complex multi modality scenario
+* Advanced caching, including periodic audits to automatically detect stale caches
+* Default ready-to-use Dataset and Sampler classes
+* See detailed introduction [here](https://github.com/IBM/fuse-med-ml/tree/master/fuse/data/README.md) 
+  
+### **fuse.eval** - a standalone library for **evaluating ML models** (not necessarily trained with FuseMedML)  
+The package includes collection of off-the-shelf metrics and utilities such as **statistical significance tests, calibration, thresholding, model comparison** and more.  
+See detailed introduction [here](https://github.com/IBM/fuse-med-ml/tree/master/fuse/eval/README.md)  
+
+## Supported DL libraries
+Some components depend on pytorch. For example, ```fuse.data``` is oriented towards pytorch DataSet, DataLoader, DataSampler etc.
+```fuse.dl``` makes heavy usage of pytorch models.
+Some components do not depend on any specific DL library - for example ```fuse.eval```.
+
+Broadly speaking, the supported DL libraries are:
+* "Pure" [pytorch](https://pytorch.org/)
+* [pytorch-lightning](https://www.pytorchlightning.ai/)
+
+Before you ask - **pytorch-lightning and FuseMedML play along very nicely and have in practice orthogonal and additive benefits :)**  
+See [Simple FuseMedML + PytorchLightning Example](./examples/fuse_examples/imaging/classification/mnist/run_mnist.py) for simple supervised learning cases, and [this example ](./examples/fuse_examples/imaging/classification/mnist/run_mnist_custom_pl_imp.py) for completely custom usage of pytorch-lightning and FuseMedML - useful for advanced scenarios such as Reinforcement Learning and generative models.
```

Comment:
In my mind this is **the most important** point and it deserves more discussion. Specifically, given that there are already a lot of DL libraries, the dev needs to be convinced to pick fuse. For example lightning addresses this question vis-a-vis vanilla torch in two ways: 
1. Via the super cool gif showing how you can transform complex vanilla torch code into simple lightning code.
2. An explicit section https://github.com/Lightning-AI/lightning#advantages-over-unstructured-pytorch

That is what is missing here. Ok, I **could** use fuse with lightning. But why add fuse if I already have fluency with lightning?

Also, lightning is not the only kid in town. There's https://github.com/project-monai/monai and probably many more that I am not familiar with. Is fuse better than all of them for every ML projects? Or are some projects better suited to vanilla torch or lightning? How should I decide? What exactly is it that fuse can do better than the others, given that everyone covers architecture, loss, training and evaluation in some way or other? Can you show me?

These are the essential questions, in my opinion.